packages/core/data-transfer/src/strapi/providers/remote-destination/index.ts

```diff
@@ -78,33 +101,48 @@ class RemoteStrapiDestinationProvider implements IDestinationProvider {
       return new ProviderTransferError('Unexpected error');
     }
 
+    this.stats[step] = { count: 0 };
+
     return null;
   }
 
   async #endStep<T extends Client.TransferPushStep>(step: T) {
     try {
-      await this.dispatcher?.dispatchTransferStep({ action: 'end', step });
+      const res = await this.dispatcher?.dispatchTransferStep<{
+        ok: boolean;
+        stats: { started: number; finished: number };
+      }>({
+        action: 'end',
+        step,
+      });
+
+      return { stats: res?.stats, error: null };
     } catch (e) {
       if (e instanceof Error) {
-        return e;
+        return { stats: null, error: e };
       }
 
       if (typeof e === 'string') {
-        return new ProviderTransferError(e);
+        return { stats: null, error: new ProviderTransferError(e) };
       }
 
-      return new ProviderTransferError('Unexpected error');
+      return { stats: null, error: new ProviderTransferError('Unexpected error') };
     }
-
-    return null;
   }
 
   async #streamStep<T extends Client.TransferPushStep>(
     step: T,
-    data: Client.GetTransferPushStreamData<T>
+    message: Client.GetTransferPushStreamData<T>
   ) {
     try {
-      await this.dispatcher?.dispatchTransferStep({ action: 'stream', step, data });
+      if (step === 'assets') {
+        const assetMessage = message as TransferAssetFlow[];
```

Comment:
So weird that it doesn't narrow it down automatically :disappointed:

Reply:
yep I tried but my typescript guru skills failed to make that happen